src/globus_cli/commands/flows/_common.py

```diff
@@ -102,3 +107,30 @@
         to create a list of keywords.
     """,
 )
+
+
+class SubscriptionIdType(click.ParamType):
+    name = "SUBSCRIPTION_ID"
+
+    def convert(
+        self, value: str, param: click.Parameter | None, ctx: click.Context | None
+    ) -> uuid.UUID | t.Literal["DEFAULT"]:
+        if value.upper() == "DEFAULT":
+            return "DEFAULT"
+        try:
+            return uuid.UUID(value)
+        except ValueError:
+            self.fail(f"{value} must be either a UUID or 'DEFAULT'", param, ctx)
+
+
+subscription_id_option = click.option(
+    "--subscription-id",
+    "subscription_id",
+    type=SubscriptionIdType(),
+    multiple=False,
```

Comment:
I would typically not specify `multiple=False`, as it's the default, but this definitely isn't harmful.